### PR TITLE
Add posible to set custom path to certificate

### PR DESCRIPTION
**Pull request by stepanLav**
### Overview
---

This PR adds an option to set custom path for certificates with grpc TLS connection.

---
### Changes

Add option to use set custom path for certificates
Add example with ssl connection using

### CODE REVIEW

**Comment by lebdron** [comment]
@ravenolf Could you check if this works in your setup?

File: examples/tls-example.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
...MIIDUDCCAjgCCQCsHEs/sUtihTANBgkqhkiG9w0BAQsFADBqMQswCQYDVQQGEwJy\ndTELMAkGA1UECAwCZGYxCzAJBgNVBAcMAmRmMQswCQYDVQQKDAJkZjELMAkGA1UE\nCwwCZGYxEjAQBgNVBAMMCWxvY2FsaG9zdDETMBEGCSqGSIb3DQEJARYEbWFpbDAe\nFw0yMDExMTkxOTM1MzVaFw0yMDEyMTkxOTM1MzVaMGoxCzAJBgNVBAYTAnJ1MQsw\nCQYDVQQIDAJkZjELMAkGA1UEBwwCZGYxCzAJBgNVBAoMAmRmMQswCQYDVQQLDAJk\nZjESMBAGA1UEAwwJbG9jYWxob3N0MRMwEQYJKoZIhvcNAQkBFgRtYWlsMIIBIjAN\nBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAu/vS/zgCIJOfpP58OFX+4EDByDnQ\n42MSQtgw1+nDQjc2thasHrNI5pFrWIRtVYUIIPhhPUEJJFW034awLWiggiZHWO9a\ngocO95AxjaborxRVavOLUIX17HnEiU6lb3lyfwq9t2bpluT+ooRM2ymDCl/NCsdS\nnvRomFvCcfpbVu0rc0hBVsLGrg+Hr/+mS88MIL6sU95Suawj8f8A0QoMN0zxQjto\nEyXVWN46vurxXMYFDbdRejwg1ba+DFajm6Dcl1nIq/u7dpG6mmDMzFVrwmYh6wYf\nXpT4eW9KNe9EEIh+loF7ohg+GE2YGXW/9f8UuQfsgyTHHj+uS0jdiD3PDwIDAQAB\nMA0GCSqGSIb3DQEBCwUAA4IBAQBQxyCYOA47Uzxc3qdYwnmqQ6oug73heh5qMbjC\ndJ9eY+BKZgd97kI+msX1ha9PKQaITONeZxE2907m2RQckCFXYFPxumQ/O8YiN3Vb\nUXy3otCPKlF+RoPEFH2UJiiIakbgE796FB/TskO1pEvdrE9i1GiCV2503VUzP9IC\nQRFMvWvZ4lk5Jw7rthL3xsdz3BDESULfN4mXQthZ98H+v7z7WaARzdOauFLwJ4rV\nXNlyBGw14951Xm/MoKQZdZlsG80xDucBf/VJrKLkQW60CWmXhit0GiZu79LzYMuN\nQAGWtoyyaJ7TmCOUR9MpH/2bna429WOhg9/hgMDGdJ9bG1Bm\n-----END CERTIFICATE-----\n'
```
Comment (ravenolf) [inline examples/tls-example.py:29]:
```suggestion
certificate = """-----BEGIN CERTIFICATE-----
MIIDUDCCAjgCCQCsHEs/sUtihTANBgkqhkiG9w0BAQsFADBqMQswCQYDVQQGEwJy
dTELMAkGA1UECAwCZGYxCzAJBgNVBAcMAmRmMQswCQYDVQQKDAJkZjELMAkGA1UE
CwwCZGYxEjAQBgNVBAMMCWxvY2FsaG9zdDETMBEGCSqGSIb3DQEJARYEbWFpbDAe
Fw0yMDExMTkxOTM1MzVaFw0yMDEyMTkxOTM1MzVaMGoxCzAJBgNVBAYTAnJ1MQsw
CQYDVQQIDAJkZjELMAkGA1UEBwwCZGYxCzAJBgNVBAoMAmRmMQswCQYDVQQLDAJk
ZjESMBAGA1UEAwwJbG9jYWxob3N0MRMwEQYJKoZIhvcNAQkBFgRtYWlsMIIBIjAN
BgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAu/vS/zgCIJOfpP58OFX+4EDByDnQ
42MSQtgw1+nDQjc2thasHrNI5pFrWIRtVYUIIPhhPUEJJFW034awLWiggiZHWO9a
gocO95AxjaborxRVavOLUIX17HnEiU6lb3lyfwq9t2bpluT+ooRM2ymDCl/NCsdS
nvRomFvCcfpbVu0rc0hBVsLGrg+Hr/+mS88MIL6sU95Suawj8f8A0QoMN0zxQjto
EyXVWN46vurxXMYFDbdRejwg1ba+DFajm6Dcl1nIq/u7dpG6mmDMzFVrwmYh6wYf
XpT4eW9KNe9EEIh+loF7ohg+GE2YGXW/9f8UuQfsgyTHHj+uS0jdiD3PDwIDAQAB
MA0GCSqGSIb3DQEBCwUAA4IBAQBQxyCYOA47Uzxc3qdYwnmqQ6oug73heh5qMbjC
dJ9eY+BKZgd97kI+msX1ha9PKQaITONeZxE2907m2RQckCFXYFPxumQ/O8YiN3Vb
UXy3otCPKlF+RoPEFH2UJiiIakbgE796FB/TskO1pEvdrE9i1GiCV2503VUzP9IC
QRFMvWvZ4lk5Jw7rthL3xsdz3BDESULfN4mXQthZ98H+v7z7WaARzdOauFLwJ4rV
XNlyBGw14951Xm/MoKQZdZlsG80xDucBf/VJrKLkQW60CWmXhit0GiZu79LzYMuN
QAGWtoyyaJ7TmCOUR9MpH/2bna429WOhg9/hgMDGdJ9bG1Bm
-----END CERTIFICATE-----
"""
```
This makes it easier to read (PEP standard)

File: iroha/iroha.py [hunk 2]
```diff
@@ -341,17 +341,27 @@ class IrohaGrpc(object):
...
-    def __init__(self, address=None, timeout=None, secure=False):
+    def __init__(self, address=None, timeout=None, secure=False, root_certificates=None,
+                  private_key=None,
+                  certificate_chain=None):
         """
         Create Iroha gRPC client
         :param address: Iroha Torii address with port, example "127.0.0.1:50051"
         :param timeout: timeout for network I/O operations in seconds
         :param secure: enable grpc ssl channel
+        :param root_certificates: The PEM-encoded root certificates as a byte string,
```
Comment (ravenolf) [inline iroha/iroha.py:352]: You can specify that without those params, grpc will use the default CA's [link](https://grpc.io/docs/guides/auth/)

**Review by ravenolf — APPROVED** [review]
LGTM :+1:

**Review by kuvaldini — APPROVED** [review]

File: examples/tls-example.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+from iroha import Iroha, IrohaGrpc
+import sys
+
+if sys.version_info[0] < 3:
+    raise Exception('Python 3 or a more recent version is required.')
+
+print("""
+
+This example works only through a TLS connection, you must first configure torii_tls_params
+https://iroha.readthedocs.io/en/master/configure/torii-tls.html.
```
Comment (baziorek) [inline examples/tls-example.py:19]: wrong link, it is not working:
```suggestion
https://iroha.readthedocs.io/en/main/configure/torii-tls.html
```

**Review by baziorek — APPROVED** [review]
I don't know if I should approve, but I've checked the changes and they looks good:)

**Comment by kuvaldini** [comment]
@baziorek If it looks good and have permissions to approve please approve

**Comment by baziorek** [comment]
> @baziorek If it looks good and have permissions to approve please approve

I've approved:
obraz
Probably my suggestion must be resolved then my approve would automatically be visible as green.

**Comment by baziorek** [comment]
@stepanLav Was it closed by accident, or on purpose? Shouldn't it be merged after corrections?

**Comment by stepanLav** [comment]
@baziorek Sorry it was closed by accident, I will recreate it

**Comment by stepanLav** [comment]
I returned all the changes back.
Sorry, I'm not a master of git commands 😅

**Review by baziorek — APPROVED** [review]
> I returned all the changes back.
> Sorry, I'm not a master of git commands sweat_smile

I'm also not the master:).

Fortunately now it should be enough to just click "Merge" somewhere.

**Pull request merged**